runtime/src/main/java/com/dylibso/chicory/runtime/Machine.java

```diff
@@ -129,8 +130,10 @@ static void eval(MStack stack, Instance instance, ArrayDeque<StackFrame> callSta
                     case IF:
                         IF(frame, stack, instruction);
                         break;
-                    case ELSE:
                     case BR:
+                        checkInterruption();
+                        // fall through
+                    case ELSE:
```

Comment:
I just noticed that this block is shared for `ELSE` and we don't need to check for interruption in that case. I flipped the order of the cases and added a fall through for `BR`.